src/app/ngx-slimscroll/directives/slimscroll.directive.ts

```diff
@@ -69,7 +70,23 @@ export class SlimScrollDirective implements OnInit, OnDestroy {
     this.mutationThrottleTimeout = 50;
   }
 
-  ngOnInit() {
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.enabled) {
+      if (this.enabled) {
+        this.setup();
+      } else {
+        this.ngOnDestroy();
```

Comment:
I don't think calling Angular's internal events is a smart idea. Just use body of `ngOnDestroy()` here please.